bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/values/TupleValueImpl.java

```diff
@@ -827,4 +829,25 @@ private void resetSize(int index) {
             size = index + 1;
         }
     }
+
+    private void validateTupleSizeAndInherentType() {
+        int numOfMandatoryTypes = this.tupleType.getTupleTypes().size();
+        // Check if tuple has at least one more members than the number of mandatory type to compensate the
+        // removed member
+        if (numOfMandatoryTypes >= this.getLength()) {
+            throw ErrorHelper.getRuntimeException(
+                    getModulePrefixedReason(ARRAY_LANG_LIB, SIZE_MISMATCH_ERROR_IDENTIFIER),
```

Comment:
From what I can understand, this is because values to mandatory types are initialised by default (in this case with the value 0) to preserve the inherent type. If we were to print the value of `tupleWithVariableInherentTypes1[1]` without doing any shifting, we would receive 0 as the output.

Reply:
My concern is about the error message. In the error message, it mentions `expected at least '3', found '2'`. "Found 2" means there should be two members in the tuple value, doesn't it?

Reply:
Yes there is a second member (in this case 0). This can easily be checked using a foreach method call and printing all the members as follows.
```
[int, int, int...] tuple1 = [1];
tuple1.forEach(function (int item) {
    io:println(item);
});
```
Output:
```
1
0
```
Note that the value for the rest parameter is not printed as it is not initialized automatically.

IMO when the user only assigns a value for the first member (1 in this case), he acknowledges the fact that the remaining members related to all the mandatory types will be automatically initialized to their respective default zero value. This has to be done because those values cannot be empty or null as it will break inherent type (this is not the case for the rest parameter however, and is therefore ignored).

Reply:
With the filler values, giving this kind of error can deviate the user from the exact reason. So, it is better to use an error message similar to "The number of members in a tuple value should be greater than the member types of the tuple", without mentioning the exact values.
Here we can use `OperationNotSupported` without specifying it as `SizeMismatch`.

Reply:
Got it. I changed the error message to something similar to what was suggested and the error reason to `OperationNotSupported`. Thank you for the feedback.

(Resolved with 333163ce157ccb8f3b4af493f2be2d358957f4cb)